### ISSUE TITLE
possible to do a tinygo version only?

### ISSUE THREAD

**Issue by kolinfluence** [maxence-charriere/go-app#787]
possible to do a tinygo version only?

**Comment by oderwat** [comment]
See: #682 and related issues. Basically, it did not create so much smaller WASM and needs a very long time to compile, the last time we tried. I have some experimental code that uses tinygo to create modules that can be loaded in addition to the main program.

**Comment by ultperf** [comment]
1. 40% or 50% or even 30% smaller wasm is a lot. worth the time to compile. no point having pwa when it loads so long and bloated.
2. do u know of any direction towards having tinygo compatibility or support? or anything other than go-app?
3. do u think future version of golang will help with minimizing the size?

**Comment by oderwat** [comment]
1. I doubt it will be much better after gzip (I actually think I did a comparison with a version I got to compile).
2. I honestly think that TinyGo is for "something else" (IOT). But a full compiler for WASM (+WASI) may come out at some day. I do not know about any other project that works like go-app. Most focus on connecting java-script libraries or even have their own front-end code which then needs a constant connection with a server.
3. It would be strange if Go would not continue to get better with WASM, as WASM is a quite active field. I believe WASM may replace (docker like) containers in the future. However: I am not sure about code size reduction (after GZip). There is a lot in such executables which seems redundant. But that is kind of a "trademark" of Go (as well as its binaries). But every time somebody wants to strip stuff, it is missing later. I hope they continue on adding debugger support. The file size becomes less important every day with those blazing fast internet the world relies on already.
